### PR TITLE
Enddat 189

### PR DESCRIPTION
**Pull request by mbucknell**
Refactored code by creating a BaseDataView which will be extended for each site collection. When extending this view, the template, panelHeading, panelBodyId, and render function will be redefined for the specific dataset. Extended this view to implement the NWISDataView, PrecipDataView and the new ACISDataView. 

Added the ACIS collection to VariableSummaryView and corrected the creation of the variableParameter for ACIS so that the process data feature works.

### CODE REVIEW

File: src/main/webapp/js/models/ACISCollection.js [hunk 1]
```diff
@@ -33,7 +34,8 @@ define([
 
 		parse : function(response) {
 			var sites = response.meta;
-			var result = _.map(sites, function(site) {
+			return _.map(sites, function(site) {
+				var sid = site.sids[0].split(' ')[0];
```
Comment (mbucknell) [inline src/main/webapp/js/models/ACISCollection.js]: The enddat-services work only if the sid is sent, so don't send the sid type.

File: src/test/js/views/NWISDataViewSpec.js [hunk 2]
```diff
@@ -89,41 +89,5 @@ define([
 
 			expect(testView.context.variables[0].selected).toBe(true);
 		});
-
```
Comment (mbucknell) [inline src/test/js/views/NWISDataViewSpec.js]: This code is part of BaseDataView and is tested in that spec.

File: src/test/js/views/PrecipDataViewSpec.js [hunk 3]
```diff
@@ -67,36 +67,5 @@ define([
 
 			expect(testView.$('input:checkbox').is(':checked')).toBe(true);
 		});
-
-		it('Expects that if the checkbox is checked, the selected property is set to true', function() {
```
Comment (mbucknell) [inline src/test/js/views/PrecipDataViewSpec.js]: The code being tested is part of BaseDataView and is tested in that spec.

File: src/main/webapp/js/views/ACISDataView.js [hunk 4]
```diff
@@ -0,0 +1,46 @@
+/* jslint browser: true */
+
+define([
+	'underscore',
+	'Config',
+	'views/BaseDataView',
+	'hbs!hb_templates/acisData'
+], function(_, Config, BaseDataView, hbTemplate) {
+
+	"use strict";
```
Comment (ayan-usgs) [inline src/main/webapp/js/views/ACISDataView.js]: What is the benefit `use strict`?

http://stackoverflow.com/questions/1335851/what-does-use-strict-do-in-javascript-and-what-is-the-reasoning-behind-it makes it sound like a few useful things, but it's unclear to me what exactly it's doing. It here to throw errors?

**Comment by ayan-usgs** [comment]
Looks good to go. :+1:

File: src/main/webapp/js/views/ACISDataView.js (hunk 4 above)
Comment (thongsav-usgs) [inline src/main/webapp/js/views/ACISDataView.js]: Sorry, eaves dropping here (and I wish I used "use strict" more often), but the biggest reason I would encourage the use of it is that it catches when you forget to declare a variable with "var". In that situation, a global variable is created and used and that's prone to side affects (other parts of the code using the same variable).

That is mentioned in one of the posts further down (access to the global object).

Comment (mbucknell) [inline src/main/webapp/js/views/ACISDataView.js]: Exactly why I use it. It catches common errors. My PyCharm IDE will actually add it for you.

**Pull request merged**